can/io/generic.py

```diff
@@ -105,5 +108,21 @@ def file_size(self) -> int:
         return self.file.tell()
 
 
+class TextIOMessageWriter(FileIOMessageWriter, metaclass=ABCMeta):
+    file: TextIO | gzip.GzipFile
```

Comment:
GzipFile returns a TextIOWrapper when opened in text mode, so this should be `TextIO` only